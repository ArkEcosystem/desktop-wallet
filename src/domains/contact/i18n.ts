export const translations: { [key: string]: any } = {
	CONTACTS_PAGE: {
		ADD_CONTACT: "Add Contact",
		ADD_CONTACT_MESSAGE: "Add your most frequent contacts for fast, easy payments",
		TITLE: "My Contacts",
		SUBTITLE: "Manage your frequent contacts",
	},
	CONTACT_FORM: {
		ADD_ADDRESS: "Add address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		DELETE_CONTACT: "Delete Contact",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
	MODAL_SEARCH_CONTACT: {
		TITLE: "Search Contact",
		DESCRIPTION: "Find the contact you need quickly and conveniently",
		PLACEHOLDER: "Search...",
	},
	VALIDATION: {
		MAXLENGTH_ERROR: "The Contact Name should have less than {{maxLength}} characters.",
		CONTACT_NAME_EXISTS: "Contact with name {{name}} already exists.",
		CONTACT_NAME_EXISTS_SUFFIX: "already exists",
	},
};
