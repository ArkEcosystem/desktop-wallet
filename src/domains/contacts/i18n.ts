export const translations: { [key: string]: any } = {
	CONTACT_FORM: {
		ADD_ADDRESS: "Add Address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
};
