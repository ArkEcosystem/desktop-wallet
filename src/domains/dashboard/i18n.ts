export const translations: { [key: string]: any } = {
	FILTER_WALLETS: {
		NETWORK: {
			TITLE: "Filter Network",
			DESCRIPTION: "Select the types of networks",
		},
		WALLETS: {
			TITLE: "Your Wallets",
			DESCRIPTION: "Select the display of your wallets",
		},
		CHART: {
			TITLE: "Portfolio Chart",
			DESCRIPTION: "You can hide this unit if you like",
		},
		TRANSACTION_HISTORY: {
			TITLE: "Transaction History",
			DESCRIPTION: "You can hide this unit if you like",
		},
	},

	TRANSACTION_HISTORY: {
		TITLE: "Transactions History",
		EMPTY_TEXT:
			"This will display the history of your transactions. But you don't have more than one transaction at the moment.",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		IMPORT: "Import",
	},

	DASHBOARD_PAGE: {
		CHART: {
			TITLE: "Portfolio Chart",
			PERCENTAGES_LABEL: "Total portfolio",
		},
	},
};
