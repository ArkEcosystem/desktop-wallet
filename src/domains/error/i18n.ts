export const translations: { [key: string]: any } = {
	APPLICATION: {
		TITLE: "Something went wrong",
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		RELOAD: "Reload",
	},
	OFFLINE: {
		TITLE: "Cannot Connect",
		DESCRIPTION: "Check your internet connection if it doesn't work, check it again.",
	},
};
