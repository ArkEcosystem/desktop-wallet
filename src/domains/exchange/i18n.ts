export const translations: { [key: string]: any } = {
	MODAL_ADD_EXCHANGE: {
		TITLE: "Exchange",
		DESCRIPTION:
			"Buy ARK directly from the ARK Desktop Wallet or through one of 20+ exchanges supplying ARK tokens using BTC, ETH or other trading pairs.",
	},
};
