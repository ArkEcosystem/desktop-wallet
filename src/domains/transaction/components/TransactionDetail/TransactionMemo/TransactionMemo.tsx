import { Icon } from "app/components/Icon";
import React from "react";
import { useTranslation } from "react-i18next";

import { TransactionDetail } from "../TransactionDetail";

type TransactionMemoProps = {
	memo: string;
};

export const TransactionMemo = ({ memo }: TransactionMemoProps) => {
	const { t } = useTranslation();

	return (
		<TransactionDetail
			label={t("TRANSACTION.SMARTBRIDGE")}
			extra={
				<div className="flex justify-center w-11 text-theme-secondary-900 dark:text-theme-secondary-600">
					<Icon name="Smartbridge" width={18} height={22} />
				</div>
			}
		>
			<p className="break-all">{memo}</p>
		</TransactionDetail>
	);
};
