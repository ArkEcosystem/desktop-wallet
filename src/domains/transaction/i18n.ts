export const translations: { [key: string]: any } = {
	ACCOUNT: "Account",
	ACCOUNT_NICKNAME: "Account Nickname",
	ADD_LINK: "Add Link",
	ADD_RECIPIENT: "Add Recipient",
	ADDRESS: "Address",
	AMOUNT: "Amount",
	BLOCK_ID: "Block ID",
	CONFIRMATIONS: "Confirmations",
	DELEGATE: "Delegate",
	DELEGATE_NAME: "Delegate Name",
	DESCRIPTION: "Description",
	ENCRYPTION_PASSWORD: "Encryption Password",
	ID: "ID",
	IPFS_HASH: "IPFS Hash",
	MNEMONIC: "Mnemonic",
	MULTIPLE: "Multiple",
	NAME: "Name",
	NETWORK: "Network",
	RECIPIENT: "Recipient",
	RECIPIENTS: "Recipients",
	RECIPIENTS_HELPTEXT: "A multiple recipient transaction allows up to {{count}} recipients in one transaction",
	REGISTRATION_TYPE: "Registration Type",
	SECOND_MNEMONIC: "2nd Mnemonic",
	SEND_ALL: "Send All",
	SENDER: "Sender",
	SIGN: "Sign",
	SIGNATURES: "Signatures",
	SINGLE: "Single",
	SINGLE_OR_MULTI: "Select a Single or Multiple Recipient Transaction",
	SMARTBRIDGE: "Smartbridge",
	TIMESTAMP: "Timestamp",
	TOTAL_AMOUNT: "Total Amount",
	TRANSACTION_FEE: "Transaction Fee",
	TRANSACTION_TYPE: "Transaction Type",
	TYPE: "Type",
	VOTER: "Voter",
	WEBSITE: "Website",
	YOUR_ADDRESS: "Your address",

	FEES: {
		LAST: "Last",
		MIN: "Min",
		AVERAGE: "Average",
	},

	TRANSACTION_TYPES: {
		BUSINESS_UPDATE: "Business Update",
		VOTE: "Vote",
	},

	MODAL_TRANSFER_DETAIL: {
		TITLE: "Transfer",
	},

	MODAL_DELEGATE_REGISTRATION_DETAIL: {
		TITLE: "Delegate Registration",
	},

	MODAL_MULTISIGNATURE_DETAIL: {
		STEP_1: {
			TITLE: "Multisignature",
		},
		STEP_2: {
			TITLE: "Multisignature Passphrase",
			DESCRIPTION: "Enter your passphrase in order to sign the Multisignature transaction.",
		},
		STEP_3: {
			TITLE: "Transaction Successful",
			DESCRIPTION: "Your signature was successfully confirmed and sent.",
		},
		WAITING_FOR_SIGNATURES: "Waiting for Signatures",
	},

	MODAL_VOTE_DETAIL: {
		TITLE: "Delegate Vote",
	},

	MODAL_IPFS_DETAIL: {
		TITLE: "Transaction IPFS Hash",
	},

	LEDGER_CONFIRMATION: {
		TITLE: "Confirm Your Transaction",
		DESCRIPTION:
			"Please review and verify the information on your Ledger device. Choose Accept to complete your transaction.",
		LOADING_MESSAGE: "Waiting for confirmation ...",
	},

	LINK_TYPES: {
		BITBUCKET: "BitBucket",
		FACEBOOK: "Facebook",
		FLICKR: "Flickr",
		GITHUB: "GitHub",
		GITLAB: "GitLab",
		LINKEDIN: "LinkedIn",
		NPM: "Npm",
		TWITTER: "Twitter",
		VIMEO: "Vimeo",
		YOUTUBE: "YouTube",
	},

	PAGE_REGISTRATION: {
		FIRST_STEP: {
			TITLE: "Registration",
			DESCRIPTION: "Select the type of registration and the address you want to register with.",
		},
		SECOND_STEP: {
			TITLE: "Register Business",
			DESCRIPTION: "Select the type of registration and the address you want to register with.",
		},
		THIRD_STEP: {
			TITLE: "Transaction Review",
			DESCRIPTION: "Check the information again before voting",
		},
	},

	PAGE_RESIGN_REGISTRATION: {
		FIRST_STEP: {
			DELEGATE: {
				TITLE: "Delegate Resignation",
				DESCRIPTION: "Resign your delegate for always.",
				WARNING:
					"Keep in mind that you cannot restore your delegate after the resignation has been registered on the blockchain.",
			},
		},
		SECOND_STEP: {
			TITLE: "Transaction Review",
			DESCRIPTION: "Check the information again before Resignation",
		},
	},

	PAGE_UPDATE_REGISTRATION: {
		FIRST_STEP: {
			BUSINESS: {
				TITLE: "Update Business",
				DESCRIPTION: "Select the type of registration and the address you want to register with.",
				WARNING:
					"Keep in mind that you cannot restore your delegate after the resignation has been registered on the blockchain.",
			},
		},
		SECOND_STEP: {
			TITLE: "Transaction Review",
			DESCRIPTION: "Check the information again before Update",
		},
	},

	PAGE_IPFS: {
		FIRST_STEP: {
			TITLE: "IPFS",
			DESCRIPTION: "Store an IPFS hash on the network.",
		},
		SECOND_STEP: {
			TITLE: "Transaction Review",
			DESCRIPTION: "Check the information again before voting",
		},
	},

	PAGE_VOTE: {
		FIRST_STEP: {
			TITLE: "Vote Transaction",
			DESCRIPTION: "Select a fee to continue",
		},
		SECOND_STEP: {
			TITLE: "Transaction Review",
			DESCRIPTION: "Check the information again before voting",
		},
	},

	PAGE_TRANSACTION_SEND: {
		FIRST_STEP: {
			TITLE: "Send",
			DESCRIPTION: "Enter details to send your money",
		},
		SECOND_STEP: {
			TITLE: "Transaction Review",
			DESCRIPTION: "Check the information again before voting",
		},
	},

	AUTHENTICATION_STEP: {
		TITLE: "Authenticate",
		DESCRIPTION: "Enter your twelve word mnemonic to authenticate the transaction.",
	},

	REPOSITORIES: {
		TITLE: "Repository",
		DESCRIPTION: "Show your projects through your repository",
	},

	SOCIAL_MEDIA: {
		TITLE: "Social Media",
		DESCRIPTION: "Tell people more about yourself through social media",
	},

	PHOTO_VIDEO: {
		TITLE: "Photo and Video",
		DESCRIPTION: "Get more users and add more information about yourself",
	},

	CONTACT_SEACH: {
		TITLE: "My addresses",
		DESCRIPTION: "Find and select preferred address from you saved wallets",
	},
};
