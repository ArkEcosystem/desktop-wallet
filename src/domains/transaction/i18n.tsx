export const translations: { [key: string]: any } = {
	MODAL_TRANSFER_DETAIL: {
		TITLE: "Transfer",
	},
};
