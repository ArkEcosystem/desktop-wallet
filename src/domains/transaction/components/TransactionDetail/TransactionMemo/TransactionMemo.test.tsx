import React from "react";
import { render } from "testing-library";

import { translations as transactionTranslations } from "../../../i18n";
import { TransactionMemo } from "./TransactionMemo";

describe("TransactionMemo", () => {
	it("should render", () => {
		const memo = "test memo";

		const { container } = render(<TransactionMemo memo={memo} />);

		expect(container).toHaveTextContent(transactionTranslations.SMARTBRIDGE);
		expect(container).toHaveTextContent(memo);

		expect(container).toMatchSnapshot();
	});
});
