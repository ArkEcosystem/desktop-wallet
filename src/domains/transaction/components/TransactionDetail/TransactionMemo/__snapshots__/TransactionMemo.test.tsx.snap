// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionMemo should render 1`] = `
<div>
  <div
    class="TransactionDetail__TransactionDetailStyled-sc-15h16fu-0 hqOyLp undefined no-ligatures"
    data-testid="TransactionDetail"
  >
    <div
      class="flex-1 space-y-2"
    >
      <div
        class="text-sm font-semibold text-theme-secondary-700 no-ligatures"
      >
        Smartbridge
      </div>
      <div
        class="font-semibold"
      >
        <p
          class="break-all"
        >
          test memo
        </p>
      </div>
    </div>
    <div
      class="flex justify-center w-11 text-theme-secondary-900 dark:text-theme-secondary-600"
    >
      <div
        class="sc-bdfBwQ bVENOE"
        height="22"
        width="18"
      >
        <svg>
          smartbridge.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;
